src/vs/workbench/contrib/terminal/common/terminalEnvironment.ts

```diff
@@ -184,7 +185,7 @@ export function getCwd(
 	logService?: ILogService
 ): string {
 	if (shell.cwd) {
-		const unresolved = (typeof shell.cwd === 'object') ? shell.cwd.fsPath : shell.cwd;
+		const unresolved = (typeof shell.cwd === 'object') ? originalFSPath(shell.cwd) : shell.cwd;
```

Comment:
Maybe just use originalFSPath here? I'm worried about regressions in the latter parts.

Reply:
I changed to just use _sanitizeCwd on the early return instead.